src/core/ext/xds/xds_client.cc

```diff
@@ -1074,6 +1157,7 @@ void XdsClient::XdsChannel::AdsCall::OnRecvMessage(absl::string_view payload) {
   {
     MutexLock lock(&xds_client()->mu_);
     if (!IsCurrentCallOnChannel()) return;
+    FallForwardIfNeededLocked();
```

Comment:
I think this should move down to line 1176, where we set `seen_response_` and set the channel's status to OK.  We don't want to do this unless we've actually gotten back a valid response.